src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,63 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AGE = 18;
+    private static final int MINIMAL_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        try {
+            checkUserDataNull(user);
+            checkLengthOfData(user);
+            checkIsUserAdult(user);
+            isLoginInStorage(user);
+        } catch (ValidationException e) {
+            throw new RuntimeException(e);
+        }
+        storageDao.add(user);
+        return user;
+    }
+
+    private User isLoginInStorage(User user) throws ValidationException {
+        for (User userInStorage:Storage.PEOPLE) {
+            if (userInStorage.getLogin().equals(user.getLogin())) {
+                throw new ValidationException("User with this login name already exist");
+            }
+        }
+        return user;
+    }
+
+    private User checkIsUserAdult(User user) throws ValidationException {
+        if (user.getAge() < MINIMAL_AGE) {
+            throw new ValidationException("Age must equals or more than "
+                    + MINIMAL_AGE);
+        }
+        return user;
+    }
+
+    private User checkLengthOfData(User user) throws ValidationException {
+        if (user.getLogin().length() < MINIMAL_LENGTH) {
+            throw new ValidationException("Login length must be more than 6");
+        }
+        if (user.getPassword().length() < MINIMAL_LENGTH) {
+            throw new ValidationException("Password length must be more than 6");
+        }
+        return user;
```

Comment:
It would look more logicaly correct if you would group checks by login, pass and age and not by length. Anyway, I`ll let the mentor decide

Reply:
agree

Reply:
> agree

I don't quite understand what you want me to do, we need to check the length of the password